boa_engine/src/value/conversions.rs

```diff
@@ -134,15 +134,15 @@ impl From<()> for JsValue {
     }
 }
 
-impl<T> From<Option<T>> for JsValue
+pub(crate) trait IntoOrUndefined {
+    fn into_or_undefined(self) -> JsValue;
+}
```

Comment:
Why do we have this trait? Is `From<Option<T>>` not sufficient?

Reply:
Thought about it and realized `From<Option<T>>` was a bit risky to have.

When I finished refactoring the return values for the `init` function of the builtins, I waited for the error messages from the compiler. However, to my surprise, the `From<Option<T>>` impl was implicitly converting the `None` returns into `JsValue::Undefined`, and I realized having that implicit conversion could cause pretty nasty bugs.
So, I essentially made it opt-in with this trait, which clearly says in the code "yeah, I really want to convert `None` into `JsValue::Undefined`"